packages/core-generator/src/expressions/component-input.ts

```diff
@@ -261,16 +262,39 @@ export class ComponentInput extends Class implements Heritable {
     return mergeTypeExpressionImports(imports);
   }
 
+  compileParentNested() {
+    const parentClass = this.heritageClauses?.[0];
+    const parentNesteds = parentClass?.members.filter(
+      (m) => m.isNested && m instanceof Property && m.initializer,
+    ).map((m) => new PropertyAssignment(
+      new Identifier(m.name),
+      new PropertyAccessChain(
+        parentClass.typeNodes?.[0]
+          || new Identifier(parentClass.defaultProps?.[0])
+          || parentClass.types?.[0].expression,
+        // A extend B - gets B identifier, decide one of which should be used
```

Comment:
I need to get the name of parent class and I don't know which way is right to do this, all three ways seem to be legit